pkg/engine2/operational_eval/vertex_resource_rule_test.go

```diff
@@ -0,0 +1,220 @@
+package operational_eval
+
+import (
+	"fmt"
+	"testing"
+
+	construct "github.com/klothoplatform/klotho/pkg/construct2"
+	knowledgebase "github.com/klothoplatform/klotho/pkg/knowledge_base2"
+	"github.com/stretchr/testify/assert"
+	gomock "go.uber.org/mock/gomock"
+)
+
+func Test_resourceRuleVertex_Key(t *testing.T) {
+	tests := []struct {
+		name string
+		v    resourceRuleVertex
+		want Key
+	}{
+		{
+			name: "resource rule vertex",
+			v: resourceRuleVertex{
+				Resource: construct.ResourceId{Name: "test"},
+				Rule: knowledgebase.AdditionalRule{
+					If: "test",
+				},
+			},
+			want: Key{
+				Ref: construct.PropertyRef{
+					Resource: construct.ResourceId{Name: "test"},
+				},
+				RuleHash: "9a510ea0226e156085c51625d22ac3679eb7530fd36ba925fcbe211da9c1b373",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert := assert.New(t)
+			got := tt.v.Key()
+			assert.Equal(tt.want, got)
```

Comment:
I'm not sure this test is useful for what `Key()` is supposed to do. A more useful test would be that two keys with different rules have different keys regardless of what those keys actually resolve to.